### PR TITLE
prov/gni: Implement ep_getopt/setopt/sizeleft

### PR DESCRIPTION
**Pull request by ztiffany**
Fixes ofi-cray/libfabric-cray#612.
Fixes ofi-cray/libfabric-cray#637.

Signed-off-by: Zach Tiffany ztiffany@cray.com

@sungeunchoi @hppritcha

### CODE REVIEW

File: prov/gni/src/gnix_ep.c [hunk 1]
```diff
@@ -1714,6 +1714,94 @@ ssize_t gnix_cancel(fid_t fid, void *context)
...
+		break;
+	case FI_CLASS_RX_CTX:
+	case FI_CLASS_SRX_CTX:
+		break;
+	default:
+		GNIX_INFO(FI_LOG_EP_CTRL, "Invalid EP type\n");
+		return -FI_EINVAL;
+	}
+
+	/* We can queue RXs indefinately, return an arbitrary low water mark. */
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_ep.c]: indefinately -> _indefinitely_

File: prov/gni/src/gnix_ep.c [hunk 2]
```diff
@@ -1714,6 +1714,94 @@ ssize_t gnix_cancel(fid_t fid, void *context)
...
+	case FI_CLASS_EP:
+		break;
+	case FI_CLASS_TX_CTX:
+		break;
+	default:
+		GNIX_INFO(FI_LOG_EP_CTRL, "Invalid EP type\n");
+		return -FI_EINVAL;
+	}
+
+	/* We can queue TXs indefinately, return an arbitrary low water mark. */
```
Comment (sungeunchoi) [inline prov/gni/src/gnix_ep.c]: ditto as above

**Comment by sungeunchoi** [comment]
:+1:

**Comment by hppritcha** [comment]
:+1:

**Pull request merged**